rust/core-lib/src/movement.rs

```diff
@@ -45,6 +45,10 @@ pub enum Movement {
     UpPage,
     /// Move down one viewport height.
     DownPage,
+    /// Move up to the next line that can preserve the cursor position.
+    UpEnforceHorizPos,
```

Comment:
Bikeshed: I might go with `UpExactPosition`?

Reply:
Naming is never Bikeshedding, after working with the code I think position seems to be implied to be Horiz position. I think your name makes more sense, will change.